packages/storybook/src/generators/configuration/lib/util-functions.ts

```diff
@@ -331,6 +444,16 @@ export function addStorybookToNamedInputs(tree: Tree) {
       );
     }
 
+    if (
+      !nxJson.targetDefaults['build-storybook'].inputs.includes(
+        '!{projectRoot}/tsconfig.storybook.json'
+      )
+    ) {
+      nxJson.targetDefaults['build-storybook'].inputs.push(
+        '!{projectRoot}/tsconfig.storybook.json'
```

Comment:
I think this should be `{projectRoot}/tsconfig.storybook.json`.

We'd want to force a rebuild if the tsconfig changes. As it stands right now, changes could be made to it without it impacting the cache, right?